extensions/typescript-language-features/src/commands/openJsDocLink.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import type * as Proto from '../protocol';
+import TypeScriptServiceClientHost from '../typeScriptServiceClientHost';
+import { Lazy } from '../utils/lazy';
+import { Command } from './commandManager';
+
+
+export class OpenJsDocLinkCommand implements Command {
+	public static readonly id = '_typescript.openJsDocLink';
+	public readonly id = OpenJsDocLinkCommand.id;
+
+	public static createCommandUri(fileSpan: Proto.FileSpan) {
+		return `command:${OpenJsDocLinkCommand.id}?${encodeURIComponent(JSON.stringify({ span: fileSpan }))}`;
+	}
+
+	public constructor(
+		private readonly lazyClientHost: Lazy<TypeScriptServiceClientHost>
+	) { }
+
+	public async execute(args: { span: Proto.FileSpan }) {
+		if (!this.lazyClientHost.hasValue) {
+			return;
+		}
+
+		const uri = this.lazyClientHost.value.serviceClient.toResource(args.span.file);
+		const doc = await vscode.workspace.openTextDocument(uri);
+		const editor = await vscode.window.showTextDocument(doc);
```

Comment:
Independent from the API: why do you use this indirection and not just a resource link, like `file:///some/path`? That would save the extra roundtrip and allow for more control in the renderer, like "cmd+alt+click" in hover could open to the side etc

Reply:
Yes I took a second look we actually don't need it for this feature right now

The TypeScript API  we originally considered required resolving the location to go to in a separate step. However TS now just returns the location directly so we can use `file` style local links. I also tested this on the web and with untitled files and both seem to work

I'll remove the isTrusted part of this change. For the future, this api would be needed if:

- We want to perform a more advanced operation when you click on a link. For example: open a file and highlight  a specific range in it

- If we ever need to lazily resolve the location to jump to

Reply:
> I'll remove the isTrusted part of this change. For the future, this api would be needed if:

Yeah, I do agree that this kind of API is useful - just wanted to keep this minimal. Also, when allowing certain command we also think about arguments of those commands. E.g there is little help in allowing a command which then does arbitrary things based on its commands.